parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileWriter.java

```diff
@@ -674,6 +694,8 @@ void writeColumnChunk(ColumnDescriptor descriptor,
     state = state.write();
     if (dictionaryPage != null) {
       writeDictionaryPage(dictionaryPage);
+    }  else if (bloomFilter != null) {
```

Comment:
My original thought is if the data is encoded in the dictionary, then there is no need to build the bloom filter because the dictionary filter can do the filtering job. But you are right that it will mix up the indexing.  Will use a separated sentence.